bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/StringListType.java

```diff
@@ -100,4 +101,30 @@ public static StringListType valueOf(String value) {
         return new StringListType(value);
     }
 
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((typeDetails == null) ? 0 : typeDetails.hashCode());
```

Comment:
I think the property `typeDetails` never will be `null` as every constructor assigns some value to it. Furthermore it would be possible to add the `final` keyword to the `typeDetails` definition in [line 36](https://github.com/openhab/openhab-core/pull/1341/files#diff-4bad14d81fc15d69e4381ecc79c0f340R36).